src/app/zap-templates/templates/app/MatterIDL.zapt

```diff
@@ -0,0 +1,80 @@
+// This IDL was generated automatically by ZAP.
+// It is for view/code review purposes only.
+
+{{#chip_shared_structs}}
+{{#if struct_is_fabric_scoped}}
+[fabric_scoped_by=k{{asUpperCamelCase struct_fabric_idx_field}}]
+{{/if}}
+struct {{name}} {
+  {{#zcl_struct_items}}
+  {{type}} k{{asUpperCamelCase label}} = {{fieldIdentifier}};
+  {{/zcl_struct_items}}
+}
+
+{{/chip_shared_structs}}
+{{!TODO: consider #chip_clusters as iteration point as well.
+{{!      Unsure about the differences}}
+{{#all_user_clusters}}
+{{side}} cluster {{asUpperCamelCase name}} = {{code}} {
+  {{#zcl_enums}}
+  enum {{asUpperCamelCase name}} : {{type}} {
+    {{#zcl_enum_items}}
+    k{{asUpperCamelCase label}} = {{value}};
+    {{/zcl_enum_items}}
+  }
+
+  {{/zcl_enums}}
+  {{#chip_cluster_specific_structs}}
+  {{#if struct_is_fabric_scoped}}
+  [fabric_scoped_by=k{{asUpperCamelCase struct_fabric_idx_field}}]
+  {{/if}}
+  struct {{name}} {
+    {{#zcl_struct_items}}
+    {{type}} k{{asUpperCamelCase label}} = {{fieldIdentifier}};
+    {{/zcl_struct_items}}
+  }
+
+  {{/chip_cluster_specific_structs}}
+  {{#zcl_events}}
+  event {{asUpperCamelCase name}}({{priority}}) = {{code}} {
+    {{#zcl_event_fields}}
+    k{{asUpperCamelCase name}} = {{fieldIdentifier}},
+    {{/zcl_event_fields}}
+  }
+
+  {{/zcl_events}}
+  {{#chip_cluster_attributes}}
+  attribute({{#if isWritableAttribute}}readonly{{else}}writable{{/if}}{{#if isReportableAttribute}}, reportable{{/if}}) {{type}} {{asLowerCamelCase name}} = {{code}};
+  {{/chip_cluster_attributes}}
+  {{#chip_cluster_commands}}
+
+  request struct {{asUpperCamelCase name}}Request {
```

Comment:
```suggestion
  request struct {{asUpperCamelCase name}}Payload {
```
perhaps?

Reply:
Is there a spec-specific term for this?

I did not know one so followed https://grpc.io/docs/what-is-grpc/core-concepts/ (unwillinly really ... because this was the IDL language more recently used) where they use Request/Response.

Also considered having the request inline as arguments types, but I saw we have a lot of them so figured structs are better.

In the future, I believe we should group request/response/command together since they are super-related with each other.

Reply:
The spec term is "the command's fields".  But I don't know that the naming here is that important in the end.

Reply:
I will approach it as lazy in this case and keep "Request" so we have request/response.